.github/workflows/validate_urls.yml

```
@@ -0,0 +1,39 @@
+name: Validate URLs
+
+on:
+  page_build:
+  schedule:
+    - cron: '00 14,23 * * *'
+  workflow_dispatch:
+    inputs:
+      branch:
+        description: 'Branch to validate'
+        default: 'main'
+        required: true
+
+jobs:
+  validate_documentation_links_on_branch:
+    name: Validate links on branch
+    runs-on: ubuntu-latest
+    env:
+      DEFAULT_BRANCH: main
+    steps:
+    - name: Configure variables
+      run: echo "GIT_BRANCH=${{ github.event.inputs.branch || env.DEFAULT_BRANCH }}" >> $GITHUB_ENV
+    - uses: urlstechie/urlchecker-action@master
+      with:
+        git_path: https://github.com/wwt/SwiftCurrent
+        branch: ${{ env.GIT_BRANCH }}
+        file_types: .md,.swift
+        exclude_urls: https://github.com/wwt/SwiftCurrent/discussions/new
+        
+  validate_documentation_links_on_website:
+    name: Validate links on Github Pages
+    runs-on: ubuntu-latest
+    steps:
+    - uses: urlstechie/urlchecker-action@master
+      with:
+        git_path: https://github.com/wwt/SwiftCurrent
+        branch: gh-pages
+        file_types: .html
```

Is it worth validating the .md files here as well? I recognize they *should* all be transpiled into HTML by the time this runs, but it's unclear to me whether there's an edge case for .md files that we could cover.

---

I don't believe so in this job.  Specifically, because I don't believe there should be any .md files, and if there are, I don't know if they are shown to the viewer.  A quick scan of the `gh-pages` in my fork does not have any such files.

---

What about CSS files? Javascript? Anything necessary there?

---

Can you link from a CSS file?  Javascript I thought about, but I don't want to open that box of javascript linking, which would be generated.  I could, and we could see if it works but I have a feeling it would break depending on how the javascript does its navigations.

---

Yeah, you can with CSS, things like background images can be links, for example.

---

There's also this, which is more relevant than backgrounds: https://developer.mozilla.org/en-US/docs/Web/CSS/url()

---

In that case, we could add `.css`.

```suggestion
        file_types: .html,.css
```

---

☝️ If that's all we need for .css scanning, commit.  But if I'm missing some feel free to make another suggestion.

---

It may be mildly overzealous but it's also cheap. I'd suggest
```suggestion
        file_types: .html,.css,.scss
```